tests/unit/onyxTest.js

```diff
@@ -617,17 +592,104 @@ describe('Onyx', () => {
-                Onyx.merge(`${ONYX_KEYS.COLLECTION.TEST_POLICY}${1}`, collectionUpdate.test_policy_1);
-                return waitForPromisesToResolve();
-            })
+
+            // WHEN mergeCollection is called with an updated collection
+            .then(() => Onyx.merge(`${ONYX_KEYS.COLLECTION.TEST_POLICY}${1}`, collectionUpdate.test_policy_1))
             .then(() => {
                 // Then we expect the callback to have called twice, once for the initial connect call + once for the collection update
-                expect(mockCallback.mock.calls.length).toBe(2);
+                expect(mockCallback).toHaveBeenCalledTimes(2);
 
                 // AND the value for the second call should be collectionUpdate
-                expect(mockCallback.mock.calls[1][0]).toEqual(collectionUpdate);
+                expect(mockCallback).toHaveBeenLastCalledWith(collectionUpdate);
+            });
+    });
+
+    it('should return a promise when set() called with the same value and there is no change', () => {
+        const promiseOne = Onyx.set('test', 'pizza');
+        expect(promiseOne).toBeInstanceOf(Promise);
+        return promiseOne
+            .then(() => {
+                const promiseTwo = Onyx.set('test', 'pizza');
+                expect(promiseTwo).toBeInstanceOf(Promise);
+            });
+    });
+
+    it('should not update a subscriber if the value in the cache has not changed at all', () => {
+        const mockCallback = jest.fn();
+        const collectionUpdate = {
+            test_policy_1: {ID: 234, value: 'one'},
+        };
+
+        // Given an Onyx.connect call with waitForCollectionCallback=true
+        connectionID = Onyx.connect({
+            key: ONYX_KEYS.COLLECTION.TEST_POLICY,
+            waitForCollectionCallback: true,
+            callback: mockCallback,
+        });
+        return waitForPromisesToResolve()
+
+            // When merge is called with an updated collection
+            .then(() => Onyx.merge(`${ONYX_KEYS.COLLECTION.TEST_POLICY}${1}`, collectionUpdate.test_policy_1))
+            .then(() => {
+                // Then we expect the callback to have called twice, once for the initial connect call + once for the collection update
+                expect(mockCallback).toHaveBeenCalledTimes(2);
+
+                // And the value for the second call should be collectionUpdate
+                expect(mockCallback).toHaveBeenNthCalledWith(2, collectionUpdate);
+            })
+
+            // When merge is called again with the same collection not modified
+            .then(() => Onyx.merge(`${ONYX_KEYS.COLLECTION.TEST_POLICY}${1}`, collectionUpdate.test_policy_1))
+            .then(() => {
+                // Then we should not expect another invocation of the callback
+                expect(mockCallback).toHaveBeenCalledTimes(2);
+            })
+
+            // WHEN merge is called again with an object of equivalent value but not the same reference
```

Comment:
```suggestion
            // When merge is called again with an object of equivalent value but not the same reference
```
There are other occurrences below